### ISSUE TITLE
Can't use Browserify?

### ISSUE THREAD

**Issue by reggi** [PixelsCommander/ReactiveElements#14]
I've been trying to find [a way to shim this to work with Browserify](http://stackoverflow.com/questions/27627544/browserify-shim-export-document-function). With no luck. It would be super rad if the module just exported a function that didn't involve `document`. Is that reasonable?

**Comment by jjzazuet** [comment]
`+1` Same issue here `:(...` thx!

**Comment by reggi** [comment]
@PixelsCommander Thoughts on this?

**Comment by PixelsCommander** [comment]
Why not just make it CJS module by adding module.export = registerReact; ? 
Will it fit? 

Or do you want a document.registerReact exactly? If so what is a motivation for this?

**Comment by reggi** [comment]
@PixelsCommander Still has `react` as a module dependancy. :(

**Comment by reggi** [comment]
@PixelsCommander realistically javascript modules have imports and exports =)

**Comment by PixelsCommander** [comment]
Exactly, you are right

Is it possible to share code you are working on now with few components and all shim stuff?

**Comment by reggi** [comment]
Here's a project I have using a custom version of reactive elements. Not sure it it helps.

https://github.com/reggi/react-component-webpack-example

**Comment by PixelsCommander** [comment]
This is exactly the fix I would like to add instead of using shims. The only difference is I am going to make require conditional.

**Comment by reggi** [comment]
I know your sick of this by now. But I can't get it to work and I don't know why. 

Here's a branch that I'm working on. 

https://github.com/reggi/react-component-webpack-example/tree/dev

It's really simple if you look at [`entry.js`](https://github.com/reggi/react-component-webpack-example/blob/dev/entry.js)

```
import countdownClass from "./countdown-class"
import registerReact from "reactive-elements"
import React from "react"
countdownClass = React.createFactory(countdownClass);
registerReact('react-countdown', countdownClass);
```

Here's the error I'm getting.

```
Uncaught ReferenceError: React is not defined
reactive-elements.js:1 Uncaught ReferenceError: React is not defined
```

**Comment by reggi** [comment]
Might this have something do do with `require` not being a browser variable? I'm not using [requirejs](http://requirejs.org/).

**Comment by PixelsCommander** [comment]
Ok, now I have your code and able to verify fixes on it. Going to deal with asap.

**Comment by PixelsCommander** [comment]
Seems that we have not a Browserify issue anymore but still there is solution you can use straightaway:

Shim React as a plugin in webpack config:

var webpack = require("webpack");

module.exports = {
    devtool: "#inline-source-map",
    entry: "./entry.js",
    output: {
        path: __dirname,
        filename: "bundle.js"
    },
    module: {
        loaders: [
            //{ test: /\.css$/, loader: "style!css" },
            {test: /\.js?$/, exclude: /node_modules/, loader: '6to5-loader?experimental&optional=selfContained'}
        ]
    },
    plugins: [
        new webpack.ProvidePlugin({
            React: "react"
        })
    ]
};

**Comment by PixelsCommander** [comment]
you may also shim registerReact and modify entry to look as simple as one line

registerReact('react-countdown', require('./countdown-class'));

it is actually not worse idea to shim react and reactive elements since web component applications could potentially include zero JS code

one more thing you could do in order to add some web components magic is to create components folder and automatically include all it contents

**Comment by PixelsCommander** [comment]
@reggi You may get fixed version on GitHub (0.4.6).

As for your code. After adding ```require('reactive-elements')('react-countdown', CountdownTimer);``` to the bottom of your timer component all you need in entry.js is ```require('./countdown-class');``` That is almost what I meant by saying "zero JS apps".

Sorry, no new versions on NPM until Thursday / Friday because of corporative NPM account and my sickness.

**Comment by reggi** [comment]
@PixelsCommander Just tested 0.4.6 and it's requiring / working.

Getting a couple of warnings though

> bind(): You are binding a component method to the component. React does this for you automatically in a high-performance way, so you can safely remove this call. See CountdownTimer [ReactCompositeComponent.js:1324]

and when I use `React.createFactory` on my element before I register it I get this error.

> Warning: This JSX uses a plain function. Only React components are valid in React's JSX transform. [warning.js:36]
